docs/src/content/docs/concepts/providers.mdx

```diff
@@ -1,14 +1,96 @@
 ---
 title: Providers
-description: GenSX architecture and design goals
+description: Learn how to manage API keys and configuration dependencies in GenSX using providers
 sidebar:
   order: 3
 ---
 
-GenSX is a simple framework for building complex generative AI and LLM workflows.
+Providers are a specialized use of [contexts](../contexts) that focus on managing configuration and dependencies for your workflow. They're built using the context system described above, but provide a more focused API for configuration management.
 
-It uses `jsx` to define and orchestrate workflows with functional, reusable components:
+The main provider available today is the `OpenAIProvider`, which manages your OpenAI API key:
 
 ```tsx
+const result = await gsx.execute(
+  <OpenAIProvider apiKey={process.env.OPENAI_API_KEY}>
+    <ChatCompletion
+      model="gpt-4o-mini"
+      messages={[{ role: "user", content: "Hello!" }]}
+    />
+  </OpenAIProvider>,
+);
+```
+
+Providers offer several benefits:
+
+- Manage complex configuration like API clients and settings
+- Provide a simpler, more focused API than raw contexts
+- Keep configuration separate from business logic
+- Enable dependency injection for testing and development
```

Comment:
I feel like this section needs a bit of an overhaul. Have you taken a look at the react provider/context docs for inspiration at all? https://react.dev/learn/passing-data-deeply-with-context

Reply:
sounds good, i will rev on this a bit

Reply:
Updated to this:

Providers offer several benefits:

- **Centralized Configuration Management** - providers manage settings like API keys and feature flags keeping configuration separate from business logic.
- **Data Sharing** - providers give you an easy way to share data between components without prop drilling to simplify your code.
- **Reusability** - providers and their associated components can easily be reused in different parts of your application.
- **Scoped State Management** - providers can be used to manage state for parts of your workflow or the entire thing.